Composer/packages/client/src/pages/setting/app-settings/TemplateFeedForm.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core';
+import { useEffect, useState, useCallback } from 'react';
+import formatMessage from 'format-message';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { firstPartyTemplateFeed } from '@bfc/shared';
+import { Link } from 'office-ui-fabric-react/lib/components/Link';
+import { useRecoilValue } from 'recoil';
+import { NeutralColors } from '@uifabric/fluent-theme/lib/fluent/FluentColors';
+import { Text } from 'office-ui-fabric-react/lib/Text';
+
+import { dispatcherState, templateFeedUrlState, templateProjectsState } from '../../../recoilModel/atoms/appState';
+import TelemetryClient from '../../../telemetry/TelemetryClient';
+
+import * as styles from './styles';
+
+const settingsContainer = css`
+  border-top: 1px solid ${NeutralColors.gray20};
+  padding-top: 5px;
+  width: 100%;
+`;
+
+const settingsText = css`
+  padding-bottom: 20px;
+  width: 100%;
+`;
+
+export const TemplateFeedForm = () => {
+  const templateFeedUrl = useRecoilValue(templateFeedUrlState);
+  const templateProjects = useRecoilValue(templateProjectsState);
+  const { setTemplateFeedUrl, fetchTemplates } = useRecoilValue(dispatcherState);
+
+  const [urlValue, setUrlValue] = useState(templateFeedUrl ? templateFeedUrl : firstPartyTemplateFeed);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [fieldDisabled, setFieldDisabled] = useState(false);
+
+  const savePendingEdits = (isFirstPartyFeed?: boolean) => {
```

Comment:
is this additional flag necessary since you are already setting the value to the default?

Reply:
Yeah its still needed, the state change hasnt happened yet on the urlValue variable within the components lifecycle even though it is set in the code prior to calling this function so the conditional on line 41 would not be truthy.

Reply:
should this maybe be a useEffect on the feed value changing then?

Reply:
no I see.. this is fine!